### PR TITLE
Sample code updates to remove dialog manager and use dialog

### PR DESCRIPTION
**Pull request by Keshav-MSFT**

### CODE REVIEW

File: samples/app-complete-auth/csharp/AppCompleteAuth/Startup.cs [hunk 1]
```diff
@@ -84,8 +84,6 @@ public void ConfigureServices(IServiceCollection services)
             // Create the Conversation state. (Used by the Dialog system itself.)
             services.AddSingleton<ConversationState>();
 
-            // Dialog Manager handles initiating the Dialog Stack, saving state, etc.
-            services.AddSingleton<DialogManager>();
```
Comment (ChetanSharma-msft) [inline samples/app-complete-auth/csharp/AppCompleteAuth/Startup.cs:88]: Hope you have tested it in any sample ?

File: samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/DialogBot.cs [hunk 2]
```diff
@@ -16,31 +16,36 @@ namespace Microsoft.BotBuilderSamples
...
         {
-            _conversationState = conversationState;
-            _userState = userState;
-            _dialog = dialog;
-            _logger = logger;
-            _dialogManager = new DialogManager(dialog) { ConversationState = conversationState, UserState = userState };
+            ConversationState = conversationState;
+            UserState = userState;
+            Dialog = dialog;
+            Logger = logger;
```
Comment (ChetanSharma-msft) [inline samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/DialogBot.cs:31]: Why we have provided the names like starting with capital letter ?
Check and rename these.

File: samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/DialogBot.cs [hunk 3]
```diff
@@ -16,31 +16,36 @@ namespace Microsoft.BotBuilderSamples
...
+        }
+
+        public override async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken = default(CancellationToken))
```
Comment (ChetanSharma-msft) [inline samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/DialogBot.cs:34]: Add Summary for all methods.

File: samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/DialogBot.cs [hunk 4]
```diff
@@ -16,31 +16,36 @@ namespace Microsoft.BotBuilderSamples
...
         }
 
         protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
         {
-            _logger.LogInformation("Running dialog with Message Activity.");
+            Logger.LogInformation("Running dialog with Message Activity.");
 
-            await _dialogManager.OnTurnAsync(turnContext, cancellationToken);
+            // Run the Dialog with the new message Activity.
+            await Dialog.RunAsync(turnContext, ConversationState.CreateProperty<DialogState>(nameof(DialogState)), cancellationToken);
```
Comment (ChetanSharma-msft) [inline samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/DialogBot.cs:48]: Why D in capital letter in Dialog ?

File: samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/TeamsBot.cs [hunk 5]
```diff
@@ -34,12 +32,12 @@ protected override async Task OnMembersAddedAsync(IList<ChannelAccount> membersA
 
         protected override async Task OnTeamsSigninVerifyStateAsync(ITurnContext<IInvokeActivity> turnContext, CancellationToken cancellationToken)
         {
-            _logger.LogInformation("Running dialog with signin/verifystate from an Invoke Activity.");
+            Logger.LogInformation("Running dialog with signin/verifystate from an Invoke Activity.");
```
Comment (ChetanSharma-msft) [inline samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/TeamsBot.cs:35]: Check and update the Logger variable

File: samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/TeamsBot.cs [hunk 6]
```diff
@@ -34,12 +32,12 @@ protected override async Task OnMembersAddedAsync(IList<ChannelAccount> membersA
...
 
             // The OAuth Prompt needs to see the Invoke Activity in order to complete the login process.
 
             // Run the Dialog with the new Invoke Activity.
-            await _dialogManager.OnTurnAsync(turnContext, cancellationToken).ConfigureAwait(false);
+            await Dialog.RunAsync(turnContext, ConversationState.CreateProperty<DialogState>(nameof(DialogState)), cancellationToken);
```
Comment (ChetanSharma-msft) [inline samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/TeamsBot.cs:40]: Check and update the Dialog variable

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added my comments.

File: samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/DialogBot.cs (hunk 2 above)
Comment (Keshav-MSFT) [reply samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/DialogBot.cs:31]: Resolved

File: samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/DialogBot.cs (hunk 3 above)
Comment (Keshav-MSFT) [reply samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/DialogBot.cs:34]: Added

File: samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/DialogBot.cs (hunk 4 above)
Comment (Keshav-MSFT) [reply samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/DialogBot.cs:48]: Resolved

File: samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/TeamsBot.cs (hunk 5 above)
Comment (Keshav-MSFT) [reply samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/TeamsBot.cs:35]: Resolved

File: samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/DialogBot.cs [hunk 7]
```diff
@@ -16,31 +16,36 @@ namespace Microsoft.BotBuilderSamples
...
         protected readonly Dialog _dialog;
         protected readonly ILogger _logger;
         protected readonly BotState _userState;
-        protected readonly DialogManager _dialogManager;
-#pragma warning restore SA1401 // Fields should be private
 
         public DialogBot(ConversationState conversationState, UserState userState, T dialog, ILogger<DialogBot<T>> logger)
         {
-            _conversationState = conversationState;
+            _conversationstate = conversationState;
```
Comment (ChetanSharma-msft) [inline samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/DialogBot.cs:28]: _conversationState

File: samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/DialogBot.cs [hunk 8]
```diff
@@ -16,31 +16,36 @@ namespace Microsoft.BotBuilderSamples
...
             _userState = userState;
             _dialog = dialog;
             _logger = logger;
-            _dialogManager = new DialogManager(dialog) { ConversationState = conversationState, UserState = userState };
         }
+        // Save any state changes that might have occurred during the turn.
```
Comment (ChetanSharma-msft) [inline samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/DialogBot.cs:33]: This is not the correct way of adding summary.

File: samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/DialogBot.cs [hunk 9]
```diff
@@ -16,31 +16,36 @@ namespace Microsoft.BotBuilderSamples
...
+        public override async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            await base.OnTurnAsync(turnContext, cancellationToken);
 
+            // Save any state changes that might have occurred during the turn.
+            await _conversationstate.SaveChangesAsync(turnContext, false, cancellationToken);
+            await _userState.SaveChangesAsync(turnContext, false, cancellationToken);
+        }
+        //  Run the Dialog with the new message Activity.
```
Comment (ChetanSharma-msft) [inline samples/bot-conversation-sso-quickstart/csharp_dotnetcore/BotConversationSsoQuickstart/Bots/DialogBot.cs:42]: This is not the correct way of adding summary.

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added my comments.

**Review by ChetanSharma-msft — APPROVED** [review]
Changes looks fine, approving it.

**Review by Wajeed-msft — APPROVED** [review]

**Pull request merged**